cpp/include/cudf/transform.hpp

```diff
@@ -224,5 +224,25 @@ std::unique_ptr<column> row_bit_count(
   table_view const& t,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns an approximate cumulative size in bits of all columns in the `table_view` for
+ * each segment of rows.
+ *
+ * This is similar to counting bit size per row for the input table, except that row sizes are
+ * accumulated by segments.
+ *
+ * Currently, only fixed-length segments are supported. In case the input table has number of rows
+ * not divisible by segment_length, its last segment is considered as shorter than the others.
+ *
+ * @param t The table view to perform the computation on
+ * @param segment_length The number of rows in each segment for which the total size is computed
+ * @param mr Device memory resource used to allocate the returned columns' device memory
+ * @return A 32-bit integer column containing the per-row bit counts
+ */
+std::unique_ptr<column> segmented_bit_count(
```

Comment:
Should this be `segmented_row_bit_count`? I think I saw this pattern in other libcudf APIs.

Reply:
@nvdbaranec How do you think?

Reply:
`segmented_row_bit_count` sounds good to me.